### ISSUE TITLE
StackOverflow

### ISSUE THREAD

**Issue by skorunka** [riteshrao/ncommon#4]
Method "public void RegisterGeneric(Type service, Type implementation)" calls its self, in StructureMapContainerAdapter.cs line 100.

```
    public void RegisterGeneric(Type service, Type implementation)
    {
        RegisterGeneric(service, implementation);
    }
```

shoud be

```
    public void RegisterGeneric(Type service, Type implementation)
    {
        Register(service, implementation);
    }
```

**Comment by yanfeng** [comment]
The same issue in UnityContainerAdapter

**Comment by riteshrao** [comment]
Fixed in commit: 11203157cbdf65da84aa145af6f023f4c5334f70

**Comment by riteshrao** [comment]
Closed.
